curl/inc/leatherman/curl/client.hpp

```diff
@@ -213,13 +262,41 @@ namespace leatherman { namespace curl {
         LEATHERMAN_CURL_NO_EXPORT void set_timeouts(context& ctx);
         LEATHERMAN_CURL_NO_EXPORT void set_write_callbacks(context& ctx);
         LEATHERMAN_CURL_NO_EXPORT void set_client_info(context &ctx);
+        LEATHERMAN_CURL_NO_EXPORT void set_client_info(std::string const& url, std::string const& file_path);
         LEATHERMAN_CURL_NO_EXPORT void set_ca_info(context& ctx);
+        LEATHERMAN_CURL_NO_EXPORT void set_ca_info(std::string const& url, std::string const& file_path);
         LEATHERMAN_CURL_NO_EXPORT void set_client_protocols(context& ctx);
 
+        template <typename ParamType> 
+        LEATHERMAN_CURL_NO_EXPORT void curl_easy_setopt_maybe(
```

Comment:
Do you want me to make it a separate commit? I can make the refactoring for the main code base (before the download_file) as one commit, then add the download_file as another commit.